File: 30-days-of-code/csharp/Day4.cs
using System;

namespace Day4
{
    class Person
    {
        // properties
        int age;

        // constructor
        Person(int age)
        {
            // check if valid age
            if (age >= 0)
            {
                this.age = age;
            }
            else
            {
                Console.WriteLine("Age is not valid, setting age to 0.");
                this.age = 0;
            }
        }

        void YearPasses()
        {
            // increment age
            age++;
        }

        string AmIOld()
        {
            // return age statement
            if (age < 13)
                return "young";
            if (age < 18)
                return "a teenager";

            return "old";
        }
    }

    class Day4
    {
        static void Main(string[] args)
        {
            // read total input numbers
            int times = Convert.ToInt32(Console.ReadLine());

            // loop number of times
            for (int i = 0; i < times; i++)
            {
                // instantiate a person
                Person p = new Person(Convert.ToInt32(Console.ReadLine()));

                // print age statement
                Console.WriteLine(p.AmIOld());

                // increment person age
                for (int j = 0; j < 3; j++)
                    p.YearPasses();

                // print age statement
                Console.WriteLine("You are {0}.", p.AmIOld());
            }
        }
    }
}

File: 30-days-of-code/java/Day4.java
import java.util.*;

class Person {
    // propertiees
    int age;

    // constructor
    Person(int age) {
        // check if valid age
        if (age >= 0) {
            this.age = age;
        } else {
            System.out.println("Age is not valid, setting age to 0.");
            this.age = 0;
        }
    }

    void yearPasses() {
        // increment age
        age++;
    }

    String amIOld() {
        // return age statement
        if (age < 13)
            return "You are young.";
        if (age < 18)
            return "You are a teenager.";

        return "You are old.";
    }
}

class Day4 {
    public static void main(String[] args) {
        // create scanner
        Scanner in = new Scanner(System.in);

        // read total input numbers
        int times = in.nextInt();

        // close scanner
        in.close();

        // loop number of times
        for (int i = 0; i < times; i++) {
            // instantiate a person
            Person p = new Person(in.nextInt());

            // print age statement
            System.out.println(p.amIOld());

            // increment person age
            for (int j = 0; j < 3; j++)
                p.yearPasses();

            // print age statement
            System.out.println(p.amIOld());
        }
    }
}
